build/build.js

```diff
@@ -64,56 +76,68 @@ const buildAllPlugin = function () {
     {name: 'gitalk', input: 'gitalk.js'}
   ]
 
-  plugins.forEach(item => {
-    build({
+  const promises = plugins.map(item => {
+    return build({
       input: 'src/plugins/' + item.input,
       output: 'plugins/' + item.name + '.js'
     })
   })
 
   if (isProd) {
     plugins.forEach(item => {
-      build({
+      promises.push(build({
         input: 'src/plugins/' + item.input,
         output: 'plugins/' + item.name + '.min.js',
         plugins: [uglify()]
-      })
+      }))
     })
   }
+
+  await Promise.all(promises)
 }
 
-if (!isProd) {
-  chokidar
-    .watch(['src/core', 'src/plugins'], {
-      atomic: true,
-      awaitWriteFinish: {
-        stabilityThreshold: 1000,
-        pollInterval: 100
-      }
-    })
-    .on('change', p => {
-      console.log('[watch] ', p)
-      const dirs = p.split(path.sep)
-      if (dirs[1] === 'core') {
-        buildCore()
-      } else if (dirs[2]) {
-        const name = path.basename(dirs[2], '.js')
-        const input = `src/plugins/${name}${
-          /\.js/.test(dirs[2]) ? '' : '/index'
-        }.js`
+async function main() {
+  if (!isProd) {
+    chokidar
+      .watch(['src/core', 'src/plugins'], {
+        atomic: true,
+        awaitWriteFinish: {
+          stabilityThreshold: 1000,
+          pollInterval: 100
+        }
+      })
+      .on('change', p => {
+        console.log('[watch] ', p)
+        const dirs = p.split(path.sep)
```

Comment:
What about the array destructuring approach?

Reply:
Sure! Feel free to make new PRs to clean up the code. I was aiming to leave code as-is except to add error handling.